pages/heft/architecture.md

```diff
@@ -0,0 +1,19 @@
+---
+layout: page
+title: Heft architecture
+navigation_source: docs_nav
+---
+
+## Terminology
+
+Heft's operation is organized around the following concepts:
+- **action** - a Heft command-line name (as formalized by Rush Stack's [ts-command-line](https://www.npmjs.com/package/@rushstack/ts-command-line) system).  For example, the "test" action appears in the  `heft test --clean` command.  Actions are the user interface, essentially macros tailored for typical developer activities.
+- **plugin** - a TypeScript class that implements the `IHeftPlugin` contract to provide arbitrary functionality.  Plugins are loaded optionally based on Heft's configuration.  You can create your own NPM package that exports custom plugins for Heft to use.  See [Core Plugins]({% link pages/heft/core_plugins.md %}) for a list of the built-in plugins.
```

Comment:
Although plain JavaScript objects are certainly possible, that seems like a practice we'd want to discourage.  Heft's API is complex and still evolving, so it might be frustrating to try to maintain an untyped plugin. Also, being "designed for TypeScript" is a pretty cool selling point compared to Webpack which has retrofitted types that aren't always super intuitive.